linera-rpc/src/grpc/server.rs

```diff
@@ -399,75 +400,133 @@ where
         cross_chain_sender_failure_rate: f32,
         cross_chain_max_concurrent_tasks: usize,
         this_shard: ShardId,
-        receiver: mpsc::Receiver<(linera_core::data_types::CrossChainRequest, ShardId)>,
+        mut receiver: mpsc::Receiver<(linera_core::data_types::CrossChainRequest, ShardId)>,
     ) {
         let pool = GrpcConnectionPool::default();
-        let max_concurrent_tasks = Some(cross_chain_max_concurrent_tasks);
-
-        receiver
-            .for_each_concurrent(max_concurrent_tasks, |(cross_chain_request, shard_id)| {
-                let shard = network.shard(shard_id);
-                let remote_address = shard.http_address();
-
-                let pool = pool.clone();
-                let nickname = nickname.clone();
-
+        let mut join_set = JoinSet::new();
```

Comment:
I don't think this affects correctness, so just a note, but I think that `tokio::task::JoinSet` won't work on the Web.  We have an abstraction `linera_core::join_set_ext::JoinSet` for this purpose.

That said this codepath is just for the server, so I'm not sure it matters.  Maybe a good practice?

Reply:
Ah, right, I hadn't even realized I was using the tokio one!